packages/fxa-auth-server/lib/firestore-db.ts

```diff
@@ -0,0 +1,32 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+import { Firestore } from '@google-cloud/firestore';
+import * as grpc from '@grpc/grpc-js';
+
+import { ConfigType } from '../config';
+
+export function setupFirestore(config: ConfigType) {
+  const fsConfig = Object.assign({}, config.authFirestore);
+  // keyFilename takes precedence over credentials
+  if (fsConfig.keyFilename) {
+    /* istanbul ignore next */
+    // @ts-ignore
+    delete fsConfig.credentials;
+  }
+
+  // Utilize the local firestore emulator when the env indicates
+  if (!!process.env.AUTH_FIRESTORE_EMULATOR_HOST) {
+    return new Firestore({
+      customHeaders: {
+        Authorization: 'Bearer owner',
+      },
+      port: 9090,
+      projectId: 'fx-event-broker',
+      servicePath: 'localhost',
+      sslCreds: grpc.credentials.createInsecure(),
+    });
+  } else {
+    return new Firestore(fsConfig);
+  }
+}
```

Comment:
I don't see tests for this module.  But since there are a couple `if`s in here, it would be nice to test the branching.

Reply:
Right, there aren't for the event-broker either. Later on there'll be integrated tests that verify the local Firestore emulator works, which effectively ensures this works correctly (as in the event-broker).